### PR TITLE
Disable the backup poller when using the background poller

### PR DESCRIPTION
**Pull request by guantaol**
Disable the backup poller in the client channel when using the
background poller.

Also, add the missing implementations of grpc_iomgr_run_in_background()
for cfstream and libuv.

### CODE REVIEW

File: src/core/ext/filters/client_channel/backup_poller.cc [hunk 1]
```diff
@@ -16,18 +16,19 @@
  *
  */
 
-#include <grpc/support/port_platform.h>
```
Comment (AspirinSJL) [inline src/core/ext/filters/client_channel/backup_poller.cc]: This should always be the first header.

Comment (guantaol) [reply src/core/ext/filters/client_channel/backup_poller.cc]: Actually clang-format moves its location. Should I follow clang-format here?

Comment (AspirinSJL) [reply src/core/ext/filters/client_channel/backup_poller.cc]: I think reverting your change here will pass the clang format check.

Comment (guantaol) [reply src/core/ext/filters/client_channel/backup_poller.cc]: To be clear, this is an auto change made by 'git clang-format', which unfortunately breaks the port platform check on github.

Comment (AspirinSJL) [reply src/core/ext/filters/client_channel/backup_poller.cc]: `git clang-format`? I believe we should use `tools/distrib/clang_format_code.sh` to format the code.

The sanity check here should definitely override `git clang-format`.

File: src/core/lib/iomgr/iomgr_uv.cc [hunk 2]
```diff
@@ -37,4 +37,6 @@ void grpc_set_default_iomgr_platform() {
   grpc_custom_iomgr_init(&grpc_uv_socket_vtable, &uv_resolver_vtable,
                          &uv_timer_vtable, &uv_pollset_vtable);
 }
+
+bool grpc_iomgr_run_in_background() { return false; }
 #endif
```
Comment (AspirinSJL) [inline src/core/lib/iomgr/iomgr_uv.cc]: This is pre-existing, but please add a blank line here.

**Comment by lidizheng** [comment]
Discussed offline. The gevent tests timeout on macOS actually is SEGFAULT. The `iomgr_posix`'s `grpc_iomgr_run_in_background` function depends on `g_event_engine` to be proper initialized, otherwise it will SEGFAULT.

To fix the gevent SEGFAULT, the solution can be a flag check similar to https://github.com/grpc/grpc/pull/18547. Alternatively, as suggested by @guantaol , can be a mocked `g_event_engine` to prevent this type of SEGFAULT in the future.

File: src/core/lib/iomgr/iomgr_uv.cc (hunk 2 above)
Comment (guantaol) [reply src/core/lib/iomgr/iomgr_uv.cc]: Done.

**Comment by guantaol** [comment]
> Discussed offline. The gevent tests timeout on macOS actually is SEGFAULT. The `iomgr_posix`'s `grpc_iomgr_run_in_background` function depends on `g_event_engine` to be proper initialized, otherwise it will SEGFAULT.
> 
> To fix the gevent SEGFAULT, the solution can be a flag check similar to #18547. Alternatively, as suggested by @guantaol , can be a mocked `g_event_engine` to prevent this type of SEGFAULT in the future.

Pushed a fix for the segfault issue, by checking whether g_event_engine is nullptr or not. g_event_engine will keep as nullptr whenever using a custom iomgr. We may use this trick in the future when having segfaults on g_event_engine with custom iomgrs (I do not fix other usages of g_event_engine in this PR).

**Comment by guantaol** [comment]
It looks that the python issue has been resolved.

Add @lidizheng as a reviewer for ev_posix.cc and @muxi as a reviewer for iomgr_posix_cfstream.cc.

The sanity check failure is a known issue #18763.

**Comment by guantaol** [comment]
Known issues:

Basic Tests Node MacOS: #17769
Basic Tests Python Windows: #19170
Bazel RBE Debug C/C++: #18840

**Pull request merged**